code/src/components/GameScreen.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import games, { GenerateQuestion } from '../reducers/games'
+import styled from 'styled-components'
+import Lottie from "lottie-react";
+import complete from "lotties/complete";
+
+
+const GameScreen = () => {
+    const description = useSelector(store => store.games.description)
+    const moves = useSelector(store => store.games.moves)
+    const history = useSelector(store => store.games.history)
+    const coordinates = useSelector(store => store.games.coordinates)
+
+    const dispatch = useDispatch()
+
+    const onChooseDirection = (event) => {
+        dispatch(games.actions.setDirection(event.target.value))
+        dispatch(GenerateQuestion())
+    }
+
+    const onRestartButton = () => {
+        dispatch(games.actions.setUserName(''))
+        window.location.reload()
+    }
+
+    const goToPreviousMove = () => {
+        dispatch(games.actions.setPreviousMove())
+    }
+
+    return (
+        <OuterWrapper>
+          <ButtonsContainer>
+            {history.length ?
+          <GoBackButton 
+            onClick={goToPreviousMove}>
+                Go back
+            </GoBackButton>
+            : <></>
+          }
+            <RestartButton
+            onClick={onRestartButton}>
+                Restart
+            </RestartButton>
+          </ButtonsContainer>
+        <InnerWrapper>
+            <h1>{description}</h1>
+            {/* <p>Your coordinates: {coordinates}</p> */}
+            {moves && moves.map(move => (
+          <InnerContainer key={move.description}>
+              <h2>{move.description}</h2>
+           <Button
+             value={move.direction} 
+             onClick={(event) => onChooseDirection(event)}>
+                 Go {move.direction}
+         </Button>
+         </InnerContainer>
+            ))}       
+          {coordinates === '1,3' && (
+            <>
+            <Lottie className="animation" animationData={complete} loop={true}/>
+            </>
+          )}
+        </InnerWrapper>
+           </OuterWrapper>
+    )
+}
+
+export default GameScreen
```

Comment:
A tip would be to rename GameScreen.js for readability to make it easier to separate it from GameBoard.js. Otherwise, the code has been easy to follow!